server/src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -309,56 +299,27 @@ public ContentAccessCertificate getCertificate(Consumer consumer)
             KeyPair keyPair = this.keyPairCurator.getConsumerKeyPair(consumer);
             byte[] pemEncodedKeyPair = pki.getPemEncoded(keyPair.getPrivate());
 
-            X509Certificate x509Cert = this.createX509Certificate(consumer, owner,
-                BigInteger.valueOf(serial.getId()), keyPair, startDate, endDate);
-
             existing = new ContentAccessCertificate();
             existing.setSerial(serial);
             existing.setKeyAsBytes(pemEncodedKeyPair);
             existing.setConsumer(consumer);
 
+            log.info("Generating new SCA payload for organization \"{}\"...", owner.getKey());
+
+            StringBuilder pem = new StringBuilder();
+            X509Certificate x509Cert = this.createX509Certificate(consumer, owner,
+                BigInteger.valueOf(serial.getId()), keyPair, startDate, endDate);
             pem.append(new String(this.pki.getPemEncoded(x509Cert)));
+            pem.append(this.createPayloadAndSignature(owner, env));
```

Comment:
Now that we're doing per-consumer-caching, let's pass in the actual consumer object here (and down the call stack of this method) so that arch filtering will also work

Reply:
Done.